src/main/java/io/gingersnapproject/cdc/RuleLifecycleManagement.java

```diff
@@ -0,0 +1,21 @@
+package io.gingersnapproject.cdc;
+
+/**
+ * Responsible for handling the rules' lifecycle.
+ *
+ * <p>A rule can be enabled/disabled as desired. These operations only work on rules that exist.
+ * Non-existent rules do not throw any exceptions.</p>
+ *
+ * <p>When disabled, the rule will flush all pending work before stopping. After stopping, the
+ * objects will remain in memory, so it is possible to enable the rule again. A rule enabled
+ * behaves the same way as a newly started rule. It connects to the database and the backend
+ * and starts publishing pending events.</p>
+ *
+ * @author Jose Bolina
+ */
+public interface RuleLifecycleManagement {
```

Comment:
Do we really need the ability to enable/disable rules at runtime?

The operator will add all rules dynamically at runtime via the ConfigMap and won't define any rules statically.

Reply:
TBH, I don't think we specifically need that. Since we destroy an engine only locally without deleting the remote data, the semantics of enabling/disabling is the same for add/remove.